src/ru/stablex/ui/widgets/Button.hx

```diff
@@ -247,19 +249,22 @@ class Button extends Text{
         if (disabled) {   //switch icon
             if (!this.disabled) {
                 this._mouseChildrenBeforeDisabled = this.mouseChildren;
+                this._formatColorBeforeDisabled = this.format.color;
             }
             this.mouseEnabled = false;
+            this.format.color = disabledFormatColor;
             this._switchIco(this._icoDisabled);
             this._switchSkin(this.skinDisabled);
         } else {
             if (this.disabled) {
                 this.mouseChildren = this._mouseChildrenBeforeDisabled;
+                this.format.color = this._formatColorBeforeDisabled;
             }
             this.mouseEnabled = true;
             this._switchIco(this._ico);
             this._switchSkin(this.skin);
         }
-
+        this.refresh();
```

Comment:
It's better to replace this line with actual application of new format to `label`